conda_libmamba_solver/shards_subset.py

```diff
@@ -257,3 +411,155 @@ def build_repodata_subset(
     log.debug("%d (channel, package) nodes discovered", len(subset.nodes))
 
     return channel_data
+
+
+# region workers
+
+
+def combine_batches_until_none(
+    in_queue: Queue[list[NodeId] | None],
+) -> Iterator[list[NodeId]]:
+    """
+    Combine lists from in_queue until we see None. Yield combined lists.
+    """
+    running = True
+    while running and (batch := in_queue.get()) is not None:
+        node_ids = batch[:]
+        with suppress(queue.Empty):
+            while running:
+                batch = in_queue.get_nowait()
+                if batch is None:
+                    # do the work but then quit
+                    running = False
+                else:
+                    node_ids.extend(batch)
```

Comment:
```suggestion
            while True:
                batch = in_queue.get_nowait()
                if batch is None:
                    # do the work but then quit
                    running = False
                    break
                else:
                    node_ids.extend(batch)
```
Instead of both loops tracking `running`, this loop could exit with break or the `Empty` exception.